code/modules/clothing/head/jobs.dm

```diff
@@ -55,6 +55,20 @@
 	desc = "There's only one man who can sniff out the dirty stench of crime, and he's likely wearing this hat."
 	icon_state = "detective"
 	armor = list(melee = 25, bullet = 5, laser = 25, energy = 10, bomb = 0, bio = 0, rad = 0)
+	var/candy_cooldown = 0
+
+/obj/item/clothing/head/det_hat/AltClick()
+	..()
+	if(ismob(loc))
+		var/mob/M = loc
+		if(candy_cooldown+120 < world.time)
+			var/obj/item/weapon/reagent_containers/food/snacks/candy_corn/CC = new /obj/item/weapon/reagent_containers/food/snacks/candy_corn(src)
+			M.put_in_hands(CC)
+			M << "You slip a candy corn from your hat."
+			candy_cooldown = world.time

```

Comment:
Make this `candy_cooldown = world.time + 120`. Then make the if up there `if(candy_cooldown < world.time)`.